app/models/manageiq/providers/amazon/cloud_manager/event_target_parser.rb

```diff
@@ -0,0 +1,122 @@
+class ManageIQ::Providers::Amazon::CloudManager::EventTargetParser
+  attr_reader :ems_event
+
+  # @param ems_event [EmsEvent] EmsEvent object
+  def initialize(ems_event)
+    @ems_event = ems_event
+  end
+
+  # Parses all targets that are present in the EmsEvent given in the initializer
+  #
+  # @return [Array] Array of ManagerRefresh::Target objects
+  def parse
+    parse_ems_event_targets(ems_event)
```

Comment:
Now that you add `ems_event` as a state (instance variable), I think you can also create `target_collection` as an instance var and let `collect_cloudwatch_api_references` operate on that.